scripts/resultsSum.pl

```diff
@@ -159,10 +159,31 @@ sub resultReporter {
 						$numOfTotal++;
 						$tapString .= "not ok " . $numOfTotal . " - " . $testName . "\n";
 						$tapString .= "  ---\n";
-						if (($diagnostic eq 'failure') || ($diagnostic eq 'all')) {
```

Comment:
See line 181.

Reply:
Can we change the logic to the following?
```
if (($diagnostic eq 'failure') || ($diagnostic eq 'all')) {
    if ($buildList =~ /jck/) {...}
    elsif ($buildList =~ /openjdk/) {...}
    else {..}
}
```

Reply:
This is to parse, filter and record the failed test case name to a formatted tap file. Only make sense for diagnotic=failure. No need to do for diagnostic=noDetail and dianostic=all

Reply:
sure, we do not need to check for dianostic=all. The current if else block at line 181 is confusing.

Reply:
Yes, the negation is normally difficult to understand. However, the way you mentioned would make the majority of code (current else block) confusing because extra if else would be needed.

Reply:
I can update to 

```
if (($diagnostic eq 'failure') && ($buildList =~ /jck/ || $buildList =~ /openjdk/)) {
} else {
}
```

The difference would be the longer code block under the `if` branch and the one-liners of an exception move to `else` branch.